Newsletters: add 158 (2021-07-21) (#608)
bitcoinops/bitcoinops.github.io: _includes/specials/taproot/en/04-why-wait.md

there's reason to be concerned about block chain reorgs---not just
accidental reorgs but also those deliberately created to take money from
early P2TR payments.

Imagine a large number of people all wanting to be one of the first to
receive a P2TR payment.  They naively send themselves some money as soon
as they see block {{ante_trb}}.[^timelocked-trb]  Those payments will be
secure in block {{site.trb}}, but they can be stolen by any miner who
creates an alternative to block {{ante_trb}}.  If the value of the money
sent to P2TR outputs is large enough, it could easily become more
profitable to attempt to mine two blocks instead of just one (see our
[fee sniping][topic fee sniping] topic for more details).

For this reason, we don't recommend your software or service generate
addresses for P2TR until you think the reorg risk has been effectively
eliminated.  We think waiting 144 blocks (approximately one day) after
activation is a reasonably conservative margin that minimizes risk
without significantly delaying you or your users from taking advantage
of the benefits of taproot.

In short:

- {{ante_trb}}: last block where anyone can spend money sent to a P2TR-style output

> - {{site.trb}}: first block where P2TR outputs can only be spent according to
>   the [BIP341][] and [BIP342][] rules.

Reviewer: ```suggestion
- {{site.trb}}: first block where P2TR outputs can only be spent if they satisfy the [BIP341][] and [BIP342][] rules.
```

(otherwise might be interpreted as being equivalent to "according to the BIP341 and BIP342 rules, this is the first block where P2TR outputs can [only] be spent")

- {{safe_trb}}: a reasonable block at which wallets can start giving their
  users [bech32m][topic bech32] receiving addresses for P2TR outputs

> - None of the above changes the advice given in the [first part][taproot

Reviewer: Should this be a bullet point? It seems weird to have three bullets that start with a block height, and then this additional note.

  series 1] of this series to enable paying to bech32m addresses as soon
  as possible.  If someone requests payment to an address for P2TR
  before you think it's safe, that's their risk to take.

[^timelocked-trb]:
    Users who want to receive a P2TR payment in the first taproot block

>     should generate an address they don't share with anyone and then

Reviewer: Even with the disclaimer at the end, this sounds a little bit like encouragement. I'd be tempted to drop this footnote.

Reviewer 2: Im tempted to keep it, personally.

Author: I don't have a strong opinion, so feel free to drop it without further input from me.  My argument in favor of keeping it is that I suspect that publishing without this information will lead to someone excitedly telling us that there's a nifty trick people can use to eliminate the danger.  In fact, AJ did exactly that when I mentioned I was writing this section (but, unknown to him, I had already written this footnote).

Reviewer: Let's keep it
